updater/lib/dependabot/updater/security_update_helpers.rb

```diff
@@ -120,13 +120,13 @@ def record_pull_request_exists_for_latest_version(checker)
         )
       end
 
-      sig { params(existing_pull_request: T::Array[T::Hash[String, String]]).void }
+      sig { params(existing_pull_request: PullRequest).void }
       def record_pull_request_exists_for_security_update(existing_pull_request)
-        updated_dependencies = existing_pull_request.map do |dep|
+        updated_dependencies = existing_pull_request.dependencies.map do |dep|
           {
-            "dependency-name": dep.fetch("dependency-name"),
-            "dependency-version": dep.fetch("dependency-version", nil),
-            "dependency-removed": dep.fetch("dependency-removed", nil)
+            "dependency-name": dep.name,
+            "dependency-version": dep.version,
+            "dependency-removed": dep.removed? ? true : nil
```

Comment:
Since this is part of a payload to the API I wanted to keep the current behavior. 

With these kinds of refactors it's easy to accidentally rewrite way too much, or introduce bugs with subtle changes. So trying to keep the payload the same!